app/controllers/secrets_controller.rb

```diff
@@ -57,21 +57,35 @@ def batch
       raise Exceptions::RecordNotFound,
             variable_ids.find { |r| !variables.map(&:id).include?(r) }
     end
-    
+
     result = {}
 
     authorize_many variables, :execute
-    
+
     variables.each do |variable|
-      unless (secret = variable.last_secret)
-        raise Exceptions::RecordNotFound, variable.resource_id
-      end
-      
-      result[variable.resource_id] = secret.value
+      result[variable.resource_id] = get_secret_from_variable variable
+
       audit_fetch variable
     end
 
     render json: result
+  rescue Encoding::UndefinedConversionError
+    raise ApplicationController::BadSecretEncoding, result
```

Comment:
This domain error should not be scoped to the `ApplicationController`.

Reply:
Should I create a separate Exceptions module for this? Similar to how [RecordNotFound](https://github.com/cyberark/conjur/blob/master/app/models/exceptions/record_not_found.rb#L4) is handled? Or is there another approach I should use?